napari/layers/_markers_layer/model.py

```diff
@@ -107,7 +107,11 @@ def coords(self, coords: np.ndarray):
         elif len(coords) > len(self._size):
             with self.freeze_refresh():
                 adding = len(coords)-len(self._size)
-                size = np.repeat([self._size[-1]], adding, axis=0)
+                if len(self._size) > 0:
+                    new_size = self._size[-1]
+                else:
+                    new_size = np.repeat(10, self._size.shape[1])
+                size = np.repeat([new_size], adding, axis=0)
```

Comment:
Can you explain what we're doing here? Are the markers in a NumPy array by default? And the array doubles in size with any new marker...? Or...?

Reply:
I have now added some comments, let me know if it is enough